Achievements/Nudist.lua

```diff
@@ -15,28 +15,59 @@ nudist_achievement.description =
 function nudist_achievement:Register(fail_function_executor)
 	nudist_achievement:RegisterEvent("PLAYER_EQUIPMENT_CHANGED")
 	nudist_achievement:RegisterEvent("PLAYER_LEVEL_UP")
+	nudist_achievement:RegisterEvent("PLAYER_ENTER_COMBAT")
+	nudist_achievement:RegisterEvent("PLAYER_ENTERING_WORLD")
+	nudist_achievement:RegisterEvent("UNIT_SPELLCAST_SUCCEEDED")
 	nudist_achievement.fail_function_executor = fail_function_executor
 end
 
 function nudist_achievement:Unregister()
 	nudist_achievement:UnregisterEvent("PLAYER_EQUIPMENT_CHANGED")
 	nudist_achievement:UnregisterEvent("PLAYER_LEVEL_UP")
+	nudist_achievement:UnregisterEvent("PLAYER_ENTER_COMBAT")
+	nudist_achievement:UnregisterEvent("PLAYER_ENTERING_WORLD")
+	nudist_achievement:UnregisterEvent("UNIT_SPELLCAST_SUCCEEDED")
 end
 
 -- Register Definitions
 nudist_achievement:SetScript("OnEvent", function(self, event, ...)
 	local arg = { ... }
 	if event == "PLAYER_EQUIPMENT_CHANGED" then
 		if arg[1] > 1 and arg[1] < 16 then -- 1 ammo, 16+ weapons
+			-- Equipped an item if true
 			if arg[2] ~= true then
-				nudist_achievement.fail_function_executor.Fail(nudist_achievement.name)
+				-- Timed for accidental equips
+				TimedNudistFailure()
 			end
 		end
-	elseif event == "PLAYER_LEVEL_UP" then
-		for i = 2, 15 do
-			if GetInventoryItemID("player", i) then
-				nudist_achievement.fail_function_executor.Fail(nudist_achievement.name)
-			end
+	elseif event == "PLAYER_LEVEL_UP" or event == "PLAYER_ENTER_COMBAT" or event == "UNIT_SPELLCAST_SUCCEEDED" then
+		-- Instant when it can affect actual gameplay
+		CheckNudistFailure()
+	elseif event == "PLAYER_ENTERING_WORLD" then
+		if HasEquipment() then
+			-- Timed when entering the world, delay so they can hear the sound
+			C_Timer.After(0, function() TimedNudistFailure() end)
 		end
 	end
 end)
+
+function HasEquipment()
+	for i = 2, 15 do
+		if GetInventoryItemID("player", i) then
+			return true
+		end
+	end
+end
+
+function CheckNudistFailure()
+	if HasEquipment() then
+		nudist_achievement.fail_function_executor.Fail(nudist_achievement.name)
+	end
+end
+
+function TimedNudistFailure()
```

Comment:
Nit: make local

Reply:
Ah making a part of the achievement works. If you tried to make it local before, it might have failed because it was defined after its use in the file

Reply:
Oh Lua has a c style definition like that? Had no idea should keep that in mind.